plugin/plaintextmodel.cpp

```diff
@@ -264,6 +277,49 @@ bool PlainTextModel::readLines(
     return false;
 }
 
+QTextCodec *PlainTextModel::detectCodec(const char *encodedCharacters, qint64 length)
+{
+    QLocale locale;
+    QTextCodec *codec = QTextCodec::codecForLocale();
+
+    UErrorCode error = U_ZERO_ERROR;
+    if (UCharsetDetector * const detector = ucsdet_open(&error)) {
+        ucsdet_setText(detector, encodedCharacters, length, &error);
+
+        int32_t count = 0;
+        if (U_FAILURE(error)) {
+            qWarning() << "Unable to detect text encoding" << u_errorName(error);
+        } else if (const UCharsetMatch ** const matches = ucsdet_detectAll(detector, &count, &error)) {
+            QTextCodec *bestMatch = nullptr;
+            QString localeName = locale.name();
+
+            for (int32_t i = 0; i < count; ++i) {
+                if (QTextCodec *detectedCodec = QTextCodec::codecForName(ucsdet_getName(matches[i], &error))) {
+                    if (detectedCodec->name() == codec->name() || 
```

Comment:
`QTextCodec::codecForName` can return NULL. Check the return value.

Reply:
I see

Reply:
I see it.

Reply:
Done

Reply:
line 297 checks if the value is NULL